src/hera/workflows/io/_io_mixins.py

```diff
@@ -117,42 +117,33 @@ def _get_inputs(cls) -> List[Union[Artifact, Parameter]]:
     def _get_as_templated_arguments(cls) -> Self:
         """Returns the Input with templated values to propagate through a DAG/Steps function."""
         object_dict = {}
-        cls_fields = get_fields(cls)
-        annotations = get_field_annotations(cls)
 
-        for field in cls_fields:
-            if param_or_artifact := get_workflow_annotation(annotations[field]):
-                if isinstance(param_or_artifact, Parameter):
-                    object_dict[field] = "{{inputs.parameters." + f"{param_or_artifact.name}" + "}}"
-                else:
-                    object_dict[field] = "{{inputs.artifacts." + f"{param_or_artifact.name}" + "}}"
-            elif not is_annotated(annotations[field]):
-                object_dict[field] = "{{inputs.parameters." + f"{field}" + "}}"
+        for field, _, annotation in _get_workflow_annotations(cls):
+            input_type = "parameters" if isinstance(annotation, Parameter) else "artifacts"
+            object_dict[field] = f"{{{{inputs.{input_type}.{annotation.name}}}}}"
```

Comment:
Slightly opinionated style suggestion for readability/human error and to match the codebase (maybe we could add a `wrap_str` function? More chars though) - using so many brackets in a row hurts 😭 
```suggestion
            object_dict[field] = "{{" + f"inputs.{input_type}.{annotation.name}" + "}}"
```